language/compiler/ir_to_bytecode/src/compiler.rs

```diff
@@ -697,8 +720,8 @@ impl<'a> Scope for ModuleScope<'a> {
         let field_map = self
             .field_definitions
             .entry(sh_idx)
-            .or_insert_with(HashMap::new);
-        field_map.insert(name.to_string(), fd_idx);
+            .or_insert_with(FieldMap::new);
+        field_map.map.push((name.to_string(), fd_idx));
```

Comment:
We could move this function to the impl of FieldMap so that we can refactor the code easier in the future in case we want to switch to a hashmap for example?

Reply:
I don't think this is relevant in the revised code.